### PR TITLE
Add a seperate controller for cluster-metadata

### PR DESCRIPTION
**Pull request by zhouke1991**
### What this PR does / why we need it
The cluster-metadata existed as a configmap in a legacy cluster, both mgmt and workload cluster.
It was used to install by ytt on the client side, but in utkg2.0 we will deprecate the way of installing resources by client side.

### CODE REVIEW

**Comment by zhouke1991** [comment]
The bootstrap controller UT failed in Main pipeline, but it passed locally.
image

File: addons/controllers/cluster_metadata_controller.go
Comment (12345lcr) [inline addons/controllers/cluster_metadata_controller.go:118]: This seems not necessary

**Review by 12345lcr — APPROVED** [review]
LGTM

**Review by wjun — APPROVED** [review]
LGTM after removing the useless codes.

**Comment by shyaamsn** [comment]
@zhouke1991 @wjun I am concerned adding more and more things to addons-manager is going to cause memory issues. I am not comfortable with this change. Lets hold on to this until we have @zjs and Steven, Yang approve this

**Comment by shyaamsn** [comment]
Thanks for the PR. Since this is a small controller the impact on memory should be less. If there are issues we can revisit it. Thanks

**Comment by zhouke1991** [comment]
@shyaamsn Thanks!

**Pull request merged**
